src/util/mayContainChildComponent.ts

```diff
@@ -0,0 +1,22 @@
+import { ElementAst } from '@angular/compiler';
+
+export function mayContainChildComponent(root: ElementAst, componentName: string, maxDepth: number = 1): boolean {
+  function traverseChildren(node: ElementAst, depth: number): boolean {
+    if (depth > maxDepth) {
```

Comment:
You can include here also the check for node children to avoid long if below:

```
if (depth > maxDepth || node.children) {
  return false;
}

...
```